src/lualib/NumberToString.ts

```diff
@@ -0,0 +1,48 @@
+// tslint:disable-next-line: variable-name
+const ____radixChars = "0123456789abcdefghijklmnopqrstuvwxyz";
+
+function __TS__NumberToString(this: number, radix?: number): string {
+    if (radix === undefined || radix === 10 || this === Infinity || this === -Infinity || this !== this) {
+        return this.toString();
+    }
+
+    radix = Math.floor(radix);
```

Comment:
Did you use some kind of reference for this implementation? If so, could you add a comment with a link.